src/init.cpp

```diff
@@ -152,64 +153,8 @@ void initializeDataJournalAndRules (
 
   enableDebugMode (rules.getBoolean ("debug"));
 
-  // The $TIMEWARRIORDB environment variable overrides the default value of ~/.timewarrior
-  Directory dbLocation;
-  char* override = getenv ("TIMEWARRIORDB");
-  dbLocation = Directory (override ? override : "~/.timewarrior");
-
-  // If dbLocation exists, but is not readable/writable/executable, error.
-  if (dbLocation.exists () &&
-      (! dbLocation.readable () ||
-       ! dbLocation.writable () ||
-       ! dbLocation.executable ()))
-  {
-    throw format ("Database is not readable at '{1}'", dbLocation._data);
-  }
-
-  // If dbLocation does not exist, ask whether it should be created.
-  bool shinyNewDatabase = false;
-
-  if (! dbLocation.exists () &&
-      (cli.getHint ("yes", false) ||
-       confirm ("Create new database in " + dbLocation._data + "?")))
-  {
-    dbLocation.create (0700);
-    shinyNewDatabase = true;
-  }
-
-  // Create extensions subdirectory if necessary.
-  Directory extensions (dbLocation);
-  extensions += "extensions";
-
-  if (! extensions.exists ())
-  {
-    extensions.create (0700);
-  }
-
-  // Create data subdirectory if necessary.
-  Directory data (dbLocation);
-  data += "data";
-
-  if (! data.exists ())
-  {
-    data.create (0700);
-  }
-
-  // Load the configuration data.
-  Path configFile (dbLocation);
-  configFile += "timewarrior.cfg";
-
-  if (! configFile.exists ())
-  {
-    File (configFile).create (0600);
-  }
-
-  rules.load (configFile._data);
-
-  // This value is not written out to disk, as there would be no point.
-  // Having located the config file, the 'db' location is already known.
-  // This is just for subsequent internal use.
-  rules.set ("temp.db", dbLocation._data);
```

Comment:
As said in other comments, `temp.db` is expected output for extensions. So I would keep this, and rather extend it for the new values `temp.extensions` and `temp.config`...